index.d.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * A ComponentStartData contains the information necessary
+ * to register a component instance with Mezzurite. This is meant
+ * to be called with the 'mezzurite/componentStart' CustomEvent
+ * as follows:
+ * @example
+ * window.dispatchEvent(new CustomEvent('mezzurite/componentStart', {
+ *  detail: ComponentStartData
+ * }));
+ * 
+ * @param id A uniquely identifiable string that corresponds to
+ * a specific instance of a registered component. Different
+ * components must have different ids, even if they have the same
+ * name. When creating an object of type ComponentStartData, it
+ * is better to use a hashing algorithm to generate a unique id,
+ * like the following:
+ * https://www.npmjs.com/package/uuid
+ * 
+ * @param name An identifier that refers to what kind of component
+ * is being tracked. It is quite possible this takes the form of
+ * the name of the class or variable that represents the component,
+ * or the name of the custom tag. Different MezzuriteComponents of
+ * the same type will have the same name.
+ */
+export interface ComponentStartData {
+  id: string;
+  name: string;
+};
+
+/**
+ * A ComponentEndData contains the information necessary to finish
+ * a component instance with Mezzurite. This is meant to be called
+ * with the 'mezzurite/componentEnd' CustomEvent as follows:
+ * @example
+ * window.dispatchEvent(new CustomEvent('mezzurite/componentEnd', {
+ *  detail: ComponentEndData
+ * }));
+ * 
+ * 
+ * @param id A uniquely identifiable string that corresponds to
+ * a specific instance of a registered component. Different
+ * components must have different ids, even if they have the same
+ * name.
+ */
+export interface ComponentEndData {
+  id: string;
+};
+
+/**
+ * A MezzuriteData contains a snapshot of all the component data within
```

Comment:
Would it be more accurate to say "A MezzuriteData object" or "A MezzuriteData instance"?

Reply:
Yeah, it's been a while since I've written this. Looking at it now, I almost feel like omitting the `Mezzurite` prefix from everything because it'd look redundant:

```ts
import { MezzuriteData } from '@microsoft/mezzurite-core';
```

Reply:
Added a more generic statement up top, as I see it elsewhere in this file.